strawberry/experimental/pydantic/conversion_types.py

```diff
@@ -35,7 +35,3 @@ def _type_definition(self) -> TypeDefinition:
     @property
     def _pydantic_type(self) -> Type[PydanticModel]:
         ...
-
-    @property
-    def _original_model(self) -> PydanticModel:
-        ...
```

Comment:
@Rubikoid sorry but i'll need to revert this change made in https://github.com/strawberry-graphql/strawberry/pull/2677

`_original_model` is not always available. It only gets set when you convert directly from pydantic. As we also support directly instantiating from the strawberry type, e.g.

```python
class User(BaseModel):
     name: str

     @staticmethod
     def foo() -> List[str]:
         return ["Patrick", "Pietro", "Pablo"]


 @strawberry.experimental.pydantic.type(model=User, all_fields=True)
 class UserType:
    ...

UserType(name="Patrick"). _original_model 
```
will cause a runtime exception.


I'll suggest for your example in the previous MR, that you do
```
@strawberry.experimental.pydantic.type(model=User, all_fields=True)
class UserType:
    @strawberry.field
    def upper_name(self: StrawberryTypeFromPydantic[User]) -> str:
        self_model = self.to_pydantic()
        return self_model.name.upper()
```
instead. It'll need to convert the model back into pydantic, which is kind of silly, but thats what you'll have to do if you want that guarantee. `to_pydantic` is a public method so that api would less likely be broken, unlike private methods.